### PR TITLE
feat: add test 7 to amb project

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Releated to #41323

### CODE REVIEW

File: curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md [hunk 1]
```diff
@@ -122,7 +122,23 @@ You can send a POST request to `/api/replies/{board}` with form data including `
 You can send a GET request to `/api/threads/{board}`. Returned will be an array of the most recent 10 bumped threads on the board with only the most recent 3 replies for each. The `reported` and `delete_password` fields will not be sent to the client.
 
 ```js
-
+async (getUserInput) => {
+  const url = getUserInput('url');
+  const checkData = await fetch(url + '/api/threads/fcc_test');
+  const parsed = await checkData.json();
+  if(!Array.isArray(parsed)) throw new Error(`An array needs to be sent to the client got ${typeof parsed}.`);
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/09-information-security/information-security-projects/anonymous-message-board.md:129]: Ideally, we should be making assertions. Not, throwing custom errors. See previous tests for general boilerplate.

We want the console to show the assertion error with specifics like:
- `expected x, but received y`

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I am still looking into what you mentioned about the use of a specific board.

**Review by ShaunSHamilton — COMMENTED** [review]
I cannot see an issue with the use of `fcc_test` as a thread - it just needs to be created for the tests.

Quite a lot needs to be done for this test you have added:
1) The board needs to exist (just check)
2) The board needs to be flushed (remove replies and threads)
3) 9 or less threads need to be added to the board, and the test needs to assert the number of threads returned
4) Threads need to be added until total equals 11 or more, and the test needs to assert the number of threads returned is 10
5) Replies need to be added to threads - same tests as thread number, but 3 instead of 10

Hope this makes sense.

**Comment by Sembauke** [comment]
This makes a lot more sense now.

**Comment by Sembauke** [comment]
Just going to close this for now until there is a better way to test.

**Pull request closed without merging**